easybuild/easyconfigs/t/Trilinos/Trilinos-12.12.1-foss-2017b-Python-2.7.14.eb

```diff
@@ -0,0 +1,41 @@
+name = 'Trilinos'
+version = '12.12.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://trilinos.sandia.gov/'
+description = """The Trilinos Project is an effort to develop algorithms and enabling technologies
+ within an object-oriented software framework for the solution of large-scale, complex multi-physics
+ engineering and scientific problems. A unique design feature of Trilinos is its focus on packages."""
+
+toolchain = {'name': 'foss', 'version': '2017b'}
+toolchainopts = {'usempi': True, 'pic': True, 'strict': True}
+
+source_urls = ['https://github.com/trilinos/Trilinos/archive/']
+sources = ['trilinos-release-12-12-1.tar.gz']
+checksums = ['5474c5329c6309224a7e1726cf6f0d855025b2042959e4e2be2748bd6bb49e18']
+
+builddependencies = [('CMake', '3.9.5')]
+
+dependencies = [
+    ('Python', '2.7.14'),
+    ('Boost', '1.65.1', versionsuffix),
+    ('SCOTCH', '6.0.4'),
+    ('SuiteSparse', '5.1.0', '-ParMETIS-4.0.3'),
+    ('ParMETIS', '4.0.3'),
+    ('netCDF', '4.5.0'),
+    ('MATIO', '1.5.11'),
+    ('GLM', '0.9.8.5'),
+    ('X11', '20171023'),
+    ('SWIG', '3.0.12', versionsuffix),
+]
+
+shared_libs = True
```

Comment:
@vanzod Any specific reason you've enabled this?

Reply:
Mainly because I building static by default made little sense to me.

Reply:
I went back to my notes and I have found this other reason I added that. Without that line, the build fails with the error below, which is caused by a missing `-lmetis` in the linker options. When using the `shared_libs = True` the linker uses `-rpath` and that is probably the reason why the error disappears.

```
[ 81%] Linking CXX executable Ifpack_CompareWithAztecOO.exe
cd /mnt/ramdisk/Trilinos/12.12.1/foss-2017b-Python-2.7.14/Trilinos-trilinos-release-12-12-1/BUILD/packages/ifpack/test/CompareWithAztecOO && /opt/easybuild/software/Compiler/GCCcore/6.4.0/CMake/3.9.5/bin/cmake -E cmake_link_script CMakeFiles/Ifpack_CompareWithAztecOO.dir/link.txt --verbose=1
/opt/easybuild/software/Compiler/GCC/6.4.0-2.28/OpenMPI/2.1.1/bin/mpicxx    -O2 -ftree-vectorize -march=native -mieee-fp -mno-recip -fPIC -std=c++11 -fopenmp -O3 -DNDEBUG  -L/opt/easybuild/software/Core/GCCcore/6.4.0/lib64 -L/opt/easybuild/software/Core/GCCcore/6.4.0/lib -L/opt/easybuild/software/Compiler/GCC/6.4.0-2.28/OpenBLAS/0.2.20/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/ScaLAPACK/2.0.2-OpenBLAS-0.2.20/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/FFTW/3.3.6/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/Python/2.7.14/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/Boost/1.65.1-Python-2.7.14/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SCOTCH/6.0.4/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/ParMETIS/4.0.3/lib -L/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/netCDF/4.5.0/lib -L/opt/easybuild/software/Compiler/GCCcore/6.4.0/MATIO/1.5.11/lib -L/opt/easybuild/software/Compiler/GCCcore/6.4.0/GLM/0.9.8.5/lib -L/opt/easybuild/software/Compiler/GCCcore/6.4.0/X11/20171023/lib -rdynamic CMakeFiles/Ifpack_CompareWithAztecOO.dir/cxx_main.cpp.o  -o Ifpack_CompareWithAztecOO.exe ../../src/libifpack.a ../../../galeri/src-xpetra/libgaleri-xpetra.a ../../../galeri/src-epetra/libgaleri-epetra.a ../../../amesos/src/libamesos.a ../../../common/auxiliarySoftware/SuiteSparse/src/libtrilinosss.a -lm /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/ScaLAPACK/2.0.2-OpenBLAS-0.2.20/lib/libscalapack.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/ParMETIS/4.0.3/lib/libparmetis.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/ParMETIS/4.0.3/lib/libmetis.a ../../../aztecoo/src/libaztecoo.a ../../../xpetra/sup/libxpetra-sup.a ../../../xpetra/src/libxpetra.a ../../../thyra/adapters/tpetra/src/libthyratpetra.a ../../../thyra/adapters/epetraext/src/libthyraepetraext.a ../../../epetraext/src/libepetraext.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/UMFPACK/Lib/libumfpack.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/CHOLMOD/Lib/libcholmod.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/COLAMD/Lib/libcolamd.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/UMFPACK/Lib/libamd.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/CCOLAMD/Lib/libccolamd.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/CAMD/Lib/libcamd.a /opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/SuiteSparse_config/libsuitesparseconfig.a ../../../triutils/src/libtriutils.a ../../../thyra/adapters/epetra/src/libthyraepetra.a ../../../thyra/core/src/libthyracore.a ../../../rtop/src/librtop.a ../../../tpetra/core/ext/libtpetraext.a ../../../tpetra/core/inout/libtpetrainout.a ../../../tpetra/core/src/libtpetra.a ../../../epetra/src/libepetra.a ../../../kokkos-kernels/src/libkokkoskernels.a ../../../kokkos/algorithms/src/libkokkosalgorithms.a ../../../kokkos/containers/src/libkokkoscontainers.a ../../../tpetra/tsqr/src/libkokkostsqr.a ../../../tpetra/classic/LinAlg/libtpetraclassiclinalg.a ../../../tpetra/classic/NodeAPI/libtpetraclassicnodeapi.a ../../../tpetra/classic/src/libtpetraclassic.a ../../../teuchos/kokkoscomm/src/libteuchoskokkoscomm.a ../../../teuchos/kokkoscompat/src/libteuchoskokkoscompat.a ../../../teuchos/remainder/src/libteuchosremainder.a ../../../teuchos/numerics/src/libteuchosnumerics.a /opt/easybuild/software/Compiler/GCC/6.4.0-2.28/OpenBLAS/0.2.20/lib/libopenblas.so /opt/easybuild/software/Core/GCCcore/6.4.0/lib64/libgfortran.so /opt/easybuild/software/Core/GCCcore/6.4.0/lib64/libgfortran.a /opt/easybuild/software/Compiler/GCC/6.4.0-2.28/OpenBLAS/0.2.20/lib/libopenblas.so /opt/easybuild/software/Core/GCCcore/6.4.0/lib64/libgfortran.so /opt/easybuild/software/Core/GCCcore/6.4.0/lib64/libgfortran.a ../../../teuchos/comm/src/libteuchoscomm.a ../../../teuchos/parameterlist/src/libteuchosparameterlist.a ../../../teuchos/core/src/libteuchoscore.a ../../../kokkos/core/src/libkokkoscore.a /usr/lib/x86_64-linux-gnu/libdl.so -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lgfortran -lquadmath 
/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/CHOLMOD/Lib/libcholmod.a(cholmod_metis.o):cholmod_metis.c:function cholmod_metis_bisector: error: undefined reference to 'METIS_ComputeVertexSeparator'
/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.1/SuiteSparse/5.1.0-ParMETIS-4.0.3/CHOLMOD/Lib/libcholmod.a(cholmod_metis.o):cholmod_metis.c:function cholmod_metis: error: undefined reference to 'METIS_NodeND'
collect2: error: ld returned 1 exit status
packages/ifpack/test/CompareWithAztecOO/CMakeFiles/Ifpack_CompareWithAztecOO.dir/build.make:148: recipe for target 'packages/ifpack/test/CompareWithAztecOO/Ifpack_CompareWithAztecOO.exe' failed
make[2]: *** [packages/ifpack/test/CompareWithAztecOO/Ifpack_CompareWithAztecOO.exe] Error 1
```

Reply:
@vanzod I've solved that issue in the easyblock, see https://github.com/easybuilders/easybuild-easyblocks/pull/1486/files#diff-48b66dba3754c937bc129f92b602b1f6R141